### DISCUSSION TITLE
Running resave/matrix-blocks with a table/supertable field with multiple rows?

### DISCUSSION THREAD

**Discussion by gbowne-quickbase** [craftcms/cms#13937] [Q&A] [answered]
Howdy! I've gotten really good at the `resave/matrix-blocks` command and have run a good number of them with some good complex logic in them. It's an amazing tool. Then I found myself scratching my head when I was trying to modify a supertable or table field in a matrix with multiple rows. 

In php, I'd use `foreach` or `array_map` to handle each item in a list, but I'm stumped because tables/supertables are not an array, and I can't quite figure out where to put the foreach, or even how to iterate. So let's say we have this simple example where we want to change the color 'red' to 'blue' for each item in this image banner component, and otherwise leave it the same.

```
./craft resave/matrix-blocks \
  --field components \
  --type "imageBanner" \
  --set bannerItems \
  --to "fn(\$block) => ['new1' => ['type' => 18, 'fields' => ['color' => (\$block->bannerItems->color == 'red') ? 'blue' : \$block-bannerItems->color ]]]
```

This will work nicely for a table/supertable that has ONE item in it. And if you run it on a table/supertable with multiple rows, it will end up with only one item in it. 

Where do we put the iterator? 
`--to "fn(\$block) => foreach(\$block->bannerItems as \$b) => [...]` ? That errors out with `error: syntax error, unexpected token "foreach", expecting ";"` and wrapping it in a `[]` or `{}` doesn't seem to fix it. 

Any guidance?

**Comment by brandonkelly** [comment, marked as answer]
Super Table provides its own `resave/supertable-blocks` command, which would be better suited for this. Unfortunately it’s not very useful at the moment. Just PR’d new options that would help (https://github.com/verbb/super-table/pull/545).
